aeon/schema/social.py

```diff
@@ -14,12 +14,11 @@ def __init__(self, pattern):
         self.pattern = pattern
         self.extension = "bin"
 
-    def read(self, file):
+    def read(self, file, ceph_proc_dir="/ceph/aeon/aeon/data/processed"):
```

Comment:
The name of `ceph_proc_dir` should be changed to something like maybe `root_to_model_dir`

Reply:
I'm still not sure about this default parameter, I think this should be solved in coordination with the issue on having a context root object. In one possible design the read method might evolve to have a default root=None or context=None parameter from which you would pull other properties, rather than having custom parameters like this.

Given this is not decided I would err on the side of not forcing a signature on read yet and just have it as a hardcoded path (or a module global if you really want to allow changing it) at this point.

Reply:
I think I'd like to keep this as a default arg for now. As a default arg, it won't impact `api.load()`, and it allows for easier testing (just set the arg to a test data path), which you can't do if it's hardcoded in the function body 

We could potentially get around this with a module global, but in addition to global modules generally being frowned upon, then we would also need to set the global in the test module, and so if we make future changes like you suggest we'd need to change something in two places, rather than just removing the default arg, right?

Reply:
Sure, sounds good it won't be much of a problem anyway. I totally agree with you on avoiding module globals, my only reasoning behind it is that currently we don't have a way of passing custom arguments into read from the load() function so if you want to use a custom root path you have to use the loader entirely from scratch.